src/OnlineVotingS.Application/Services/ImplService/CampaignService.cs

```diff
@@ -12,71 +13,176 @@
+    {
+        _logger.LogInformation("Fetching all campaigns.");
+
+        try
+        {
+            var campaigns = await _campaignRepository.GetAllAsync();
+            _logger.LogInformation("Fetched {Count} campaigns.", campaigns.Count());
+            return campaigns;
+        }
+        catch (Exception ex)
         {
-            return await _campaignRepository.GetAllAsync();
+            _logger.LogError(ex, "An error occurred while fetching all campaigns.");
+            throw;
         }
+    }
+
+    public async Task<IEnumerable<Campaign>> GetCampaignsByElectionIdAsync(int electionId)
+    {
+        _logger.LogInformation("Fetching campaigns for election ID {ElectionId}.", electionId);
 
-        public async Task<IEnumerable<Campaign>> GetCampaignsByElectionIdAsync(int electionId)
+        try
         {
-            return await _campaignRepository.GetByElectionIdAsync(electionId);
+            var campaigns = await _campaignRepository.GetByElectionIdAsync(electionId);
+            _logger.LogInformation("Fetched {Count} campaigns for election ID {ElectionId}.", campaigns.Count(), electionId);
+            return campaigns;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while fetching campaigns for election ID {ElectionId}.", electionId);
+            throw;
+        }
+    }
 
-        public async Task<IEnumerable<Campaign>> GetCampaignsByCandidateIdAsync(int candidateId)
+    public async Task<IEnumerable<Campaign>> GetCampaignsByCandidateIdAsync(int candidateId)
+    {
+        _logger.LogInformation("Fetching campaigns for candidate ID {CandidateId}.", candidateId);
+
+        try
+        {
+            var campaigns = await _campaignRepository.GetByCandidateIdAsync(candidateId);
+            _logger.LogInformation("Fetched {Count} campaigns for candidate ID {CandidateId}.", campaigns.Count(), candidateId);
+            return campaigns;
+        }
+        catch (Exception ex)
         {
-            return await _campaignRepository.GetByCandidateIdAsync(candidateId);
+            _logger.LogError(ex, "An error occurred while fetching campaigns for candidate ID {CandidateId}.", candidateId);
+            throw;
         }
+    }
+
+    public async Task<IEnumerable<Campaign>> GetActiveCampaignsAsync()
+    {
+        _logger.LogInformation("Fetching all active campaigns.");
 
-        public async Task<IEnumerable<Campaign>> GetActiveCampaignsAsync()
+        try
+        {
+            var activeCampaigns = await _campaignRepository.GetActiveCampaignsAsync();
+            _logger.LogInformation("Fetched {Count} active campaigns.", activeCampaigns.Count());
+            return activeCampaigns;
+        }
+        catch (Exception ex)
         {
-            return await _campaignRepository.GetActiveCampaignsAsync();
+            _logger.LogError(ex, "An error occurred while fetching active campaigns.");
```

Comment:
follow please if it works the same guidline as I mentonied here https://github.com/ArianitSGashi/OnlineVotingS/pull/45/files#r1723195909